etc/telegraf.conf

```diff
@@ -54,6 +54,13 @@
   ## ie, a jitter of 5s and interval 10s means flushes will happen every 10-15s
   flush_jitter = "0s"
 
+	## FlushAfterFirstIntervalCollection instructs Telegraf to flush metrics after
+	## first starting and collecting the first round of metrics. This is
+	## specifically used for cases where your FlushInterval is really large, but
+	## you don't want to wait the full FlushInterval to see your first metrics
+	## come through. It's typically not required to set this.
+	# flush_after_first_interval_collection = false
```

Comment:
We generally avoid adding true/false settings and settings that are essentially "make telegraf work for a specific user".  If we accepted changes of this type for every user request there would be too many settings.  Too many settings make it more confusing for everyone, therefore less useful for everyone.  The code also becomes full of tiny special case branches and harder to maintain and test.

I think we should put some thought into meeting this user request without a setting.  I suspect most users wouldn't care if there was an initial quick gather before normal intervals start.

If we can't find a way to make this work without a true/false setting we may need to acknowledge that telegraf isn't the right tool for the job and suggesting that the user fork and build a custom telegraf with this change.

Reply:
I totally agree that using booleans to customize code paths is the road to madness, but I don't think there's anything inherently bad about booleans, as long as they couldn't be better represented by another type.

I think the above boolean doesn't convey intent very well. What I really want are two things:
```
collect_metrics_immediately_on_start = true
must_flush_metrics_on_start_after = "2m"
```
and `must_flush_metrics_on_start_after` is a symptom of the outputs not knowing how long to wait to make sure every input has reported all their metrics (and as far as i know has no way to tell).

This should give the right level of detail to configure it to any scenario, and not any one specific user.

If you look at what https://github.com/influxdata/telegraf/issues/7293 is asking, the issue is that `interval` and `flush_interval`s are not enough dials to tune for reasonable behavior when using high values for these fields for infrequently-collected and infrequently-uploaded metrics.